sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveMetastoreCatalog.scala

```diff
@@ -273,6 +273,22 @@ private[hive] class HiveMetastoreCatalog(val client: ClientInterface, hive: Hive
         serdeProperties = options)
     }
 
+    def hasPartitionColumns(relation: BaseRelation): Boolean = relation match {
+      case hadoopFsRelation: HadoopFsRelation =>
+        try {
+          // Calling hadoopFsRelation.partitionColumns will trigger the refresh call of
+          // the HadoopFsRelation, which will validate input paths. However, when we create
+          // an empty table, the dir of the table has not been created, which will
+          // cause a FileNotFoundException. So, at here we will catch the FileNotFoundException
+          // and return false.
+          hadoopFsRelation.partitionColumns.nonEmpty
+        } catch {
+          case _: java.io.FileNotFoundException =>
+            false
+        }
+      case _ => false
+    }
```

Comment:
This function is equivalent with `val resolvedRelation = dataSource.resolveRelation(checkPathExist = false)` in 2.0 (https://github.com/apache/spark/blob/master/sql/core/src/main/scala/org/apache/spark/sql/execution/command/createDataSourceTables.scala#L427).